build/Dockerfile.olm-registry

```diff
@@ -1,10 +1,10 @@
-FROM registry.redhat.io/openshift4/ose-operator-registry-rhel9:v4.18.0 AS builder
+FROM registry.redhat.io/openshift4/ose-operator-registry:v4.14 AS builder
```

Comment:
We shipped this as rhel9 out of band from boilerplate to test/prove it works. We have a ticket in flight to update the image in boilerplate so it's fine this is reverting to 8 for now